adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralBannerLoader.m

```diff
@@ -0,0 +1,158 @@
+    if (!adUnitId.length || !placementId.length) {
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"Ad Unit ID or Placement ID cannot be nil.");
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
+
+    NSError *error = nil;
+    CGSize bannerSize = [self bannerSizeFromGADAdSize:adConfiguration.adSize error:&error];
+    if (error) {
+      _adLoadCompletionHandler(nil, error);
+      return;
+    }
+
+    _bannerAdView = [[MTGBannerAdView alloc]initBannerAdViewWithAdSize:bannerSize placementId:placementId unitId:adUnitId rootViewController:rootViewController];
+    _bannerAdView.delegate = self;
+    _bannerAdView.autoRefreshTime = 0;
+    [_bannerAdView loadBannerAdWithBidToken:adConfiguration.bidResponse];
+}
+
+- (CGSize)bannerSizeFromGADAdSize:(GADAdSize)gadAdSize error:(NSError **)error {
+    CGSize gadAdCGSize = CGSizeFromGADAdSize(gadAdSize);
+    GADAdSize banner50 = GADAdSizeFromCGSize(
+                                             CGSizeMake(gadAdCGSize.width, mintegralBannerAdSize320x50.height));  // 320*50
+    GADAdSize banner100 = GADAdSizeFromCGSize(
+                                              CGSizeMake(gadAdCGSize.width, mintegralBannerAdSize320x100.height));  // 320*100
+    GADAdSize banner250 = GADAdSizeFromCGSize(
+                                              CGSizeMake(gadAdCGSize.width, mintegralBannerAdSize300x250.height));  // 300*250
+    GADAdSize banner90 = GADAdSizeFromCGSize(
+                                             CGSizeMake(gadAdCGSize.width, mintegralBannerAdSize728x90.height));  // 728*90
+    NSArray *potentials = @[
+        NSValueFromGADAdSize(banner50),NSValueFromGADAdSize(banner100), NSValueFromGADAdSize(banner90), NSValueFromGADAdSize(banner250)
+    ];
+    GADAdSize closestSize = GADClosestValidSizeForAdSizes(gadAdSize, potentials);
+    CGSize size = CGSizeFromGADAdSize(closestSize);
+    if (size.height == mintegralBannerAdSize320x50.height) {
+        return mintegralBannerAdSize320x50;
+    }else if (size.height == mintegralBannerAdSize320x100.height) {
+        return mintegralBannerAdSize320x100;
+    }else if (size.height == mintegralBannerAdSize300x250.height) {
+        return mintegralBannerAdSize300x250;
+    }else if (size.height == mintegralBannerAdSize728x90.height) {
+        return mintegralBannerAdSize728x90;
+    }
+    
+    if (error) {
+        *error = GADMAdapterMintegralErrorWithCodeAndDescription(
+                                                                 GADMintegtalErrorBannerSizeMismatch,
+                                                                 [NSString stringWithFormat:@"Invalid size for Mintegral mediation adapter. Size: %@",
+                                                                  NSStringFromGADAdSize(gadAdSize)]);
+    }
+    return CGSizeZero;
+}
```

Comment:
We should update this method to below:
```
static CGSize const mintegralBannerAdSize320x50 = (CGSize){320, 50};
static CGSize const mintegralBannerAdSize320x100 = (CGSize){320, 100};
static CGSize const mintegralBannerAdSize300x250 = (CGSize){300, 250};
static CGSize const mintegralBannerAdSize728x90 = (CGSize){728, 90};


/// Returns the closest valid banner size by comparing the provided ad size against the valid sizes.
/// Returns CGRectZero and sets |error| if the ad configuration contains an invalid ad size.
+ (CGRect)bannerSizeFromAdConfiguration:(nonnull GADMediationBannerAdConfiguration *)adConfiguration
                                  error:(NSError **)errorPtr {
  GADAdSize adSize320x50 = GADAdSizeFromCGSize(CGSizeMake(320, 50));
  GADAdSize adSize320x100 = GADAdSizeFromCGSize(CGSizeMake(320, 100));
  GADAdSize adSize300x250 = GADAdSizeFromCGSize(CGSizeMake(300, 250));
  GADAdSize adSize728x90 = GADAdSizeFromCGSize(CGSizeMake(728, 90));
  NSArray<NSValue *> *possibleSizes = @[ @(adSize320x50), @(adSize320x100),  @(adSize300x250), @(adSize728x90)];

  GADAdSize requestedSize = adConfiguration.adSize;
  GADAdSize closestAdSize = GADClosestValidSizeForAdSizes(requestedSize, possibleSizes);

  if (GADAdSizeEqualToSize(closestAdSize, GADAdSizeInvalid)) {
    NSString *errorMessage = [NSString
        stringWithFormat:
            @"The requested banner size: %@ is not supported by Mintegral SDK.",
            NSStringFromGADAdSize(requestedSize)];
    *errorPtr = GADMAdapterMintegralErrorWithCodeAndDescription(
        GADMintegtalErrorBannerSizeMismatch, errorMessage);
    return CGRectZero;
  }

  return CGRectMake(0, 0, closestAdSize.size.width, closestAdSize.size.height);
}


```

Reply:
thanks, i have fixed it.

But here should return the CGSize type.